distributed/core.py

```diff
@@ -866,33 +866,47 @@ async def handle_stream(self, comm, extra=None):
             await comm.close()
             assert comm.closed()
 
-    async def close(self, timeout=None):
-        try:
-            for pc in self.periodic_callbacks.values():
-                pc.stop()
-
-            if not self.__stopped:
-                self.__stopped = True
-                _stops = set()
-                for listener in self.listeners:
-                    future = listener.stop()
-                    if inspect.isawaitable(future):
-                        warnings.warn(
-                            f"{type(listener)} is using an asynchronous `stop` method. "
-                            "Support for asynchronous `Listener.stop` will be removed in a future version",
-                            PendingDeprecationWarning,
-                        )
-                        _stops.add(future)
-                if _stops:
-                    await asyncio.gather(*_stops)
+    async def close_unsafe(
+        self, timeout: float | None, reason: str | None, **kwargs: Any
+    ) -> None:
```

Comment:
I'm not happy with the signature but that's the way it is. Specifically the Nanny and the Worker have very custom signatures making this a bit ugly